cas-server-support-osf/src/main/java/io/cos/cas/web/flow/OpenScienceFrameworkLoginHandler.java

```diff
@@ -139,22 +150,29 @@ public Event beforeLogin(final RequestContext context) {
         final OpenScienceFrameworkLoginContext osfLoginContext;
         final String serviceUrl = getEncodedServiceUrl(context);
         final boolean institutionLogin = isInstitutionLogin(context);
+        final boolean orcidRedirect = isOrcidRedirect(context);
 
         String jsonLoginContext = (String) context.getFlowScope().get("jsonLoginContext");
         if (jsonLoginContext == null) {
-            // Create a new login context
+            // Create a new login context with service URL and institution login flag
             osfLoginContext = new OpenScienceFrameworkLoginContext(serviceUrl, institutionLogin);
+            // Only allow ORCiD login redirect from a brand new login flow
+            osfLoginContext.setOrcidRedirect(orcidRedirect);
```

Comment:
As a quick hot fix, only allow this behavior for a brand new flow. Please improve.